crate_universe/src/cli/splice.rs

```diff
@@ -81,13 +85,14 @@ pub fn splice(opt: SpliceOptions) -> Result<()> {
     };
 
     // Generate a splicer for creating a Cargo workspace manifest
-    let splicer = Splicer::new(splicing_dir, splicing_manifest)?;
+    let splicer = Splicer::new(splicing_dir.clone(), splicing_manifest)?;
+    let prepared_splicer = splicer.prepare()?;
```

Comment:
It doesn't feel like `Splicer` is worth existing, if all it's used for is to construct a builder to turn into a `SplicerKind` - how would you feel about inlining `Splicer::new` into `SplicerKind::new`?

Reply:
Oh also feel free to not act on this - I can see the lifetimes here probably make this fiddly, we can pick this up separately if useful :)

Reply:
yeah, the lifetimes were annoying. I'll give it another shot but if I can't figure it out after a bit of poking I'll punt on this bit for now